docs/semgrep-supply-chain/getting-started.md

```diff
@@ -38,8 +38,11 @@ Semgrep Supply Chain is **automatically enabled** for all repositories that you
 
 When the scan finishes, Semgrep Cloud Platform displays an overview of findings in the **Supply Chain** page for further triage and remediation. See [Triaging and remediating dependency findings](/semgrep-supply-chain/triage-and-remediation).
 
+### Warning on Project Directory Structure
+Semgrep Supply Chain assumes your repository contains a [lockfile](/semgrep-supply-chain/glossary/#lockfile). It also assumes that all code files using the dependencies in this lockfile are nested in the same directory as the lockfile, potentially within deeper directories. For example, if your lockfile is in `my-project/running/lockfile.json`, Semgrep Supply Chain assumes that all code files using the dependencies in `my-project/running/lockfile.json` are nested in `my-project/running/` or deeper directories. If you have code files in `my-project/biking`, Semgrep Supply Chain will not associate them to the dependencies in `my-project/running/lockfile.json`. If there is another lockfile in `my-project/running`, like `my-project/running/uphill/lockfile.json`, then this will override the original `my-project/running/lockfile.json` for all code files in `my-project/running/uphill/` or deeper directories.
```

Comment:
I think this is probably feeling wordy because "in the same directory, potentially within deeper directories" is a pretty long way to describe this.

The description I used in the KB article is:

> Semgrep Supply Chain searches the parent directories of any code files for the nearest relevant lockfile. Monolithic repositories (monorepos) have their findings grouped based on the lockfiles present in subdirectories.

I think wording similar to that would work here, along with your example. Your example might be better as or supplemented with a visual or ASCII representation of the directory tree and notations of which code goes to which lockfiles, though. As much as I like words, they aren't great for descriptions of non-linear things like tree structures. :)